kernel/interrupts/src/aarch64/mod.rs

```diff
@@ -0,0 +1,416 @@
+        writeln!(f, "\r            Carry    (C): {}", to_flag_str(self.0.is_set(SPSR_EL1::C)))?;
+        writeln!(f, "\r            Overflow (V): {}", to_flag_str(self.0.is_set(SPSR_EL1::V)))?;
+
+        let to_mask_str = |x| -> _ { if x { "Masked" } else { "Unmasked" } };
+
+        writeln!(f, "\r      Exception handling state:")?;
+        writeln!(f, "\r            Debug  (D): {}", to_mask_str(self.0.is_set(SPSR_EL1::D)))?;
+        writeln!(f, "\r            SError (A): {}", to_mask_str(self.0.is_set(SPSR_EL1::A)))?;
+        writeln!(f, "\r            IRQ    (I): {}", to_mask_str(self.0.is_set(SPSR_EL1::I)))?;
+        writeln!(f, "\r            FIQ    (F): {}", to_mask_str(self.0.is_set(SPSR_EL1::F)))?;
+
+        write!(f, "\r      Illegal Execution State (IL): {}",
+            to_flag_str(self.0.is_set(SPSR_EL1::IL))
+        )
+    }
+}
+
+#[rustfmt::skip]
+impl fmt::Display for EsrEL1 {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        // Raw print of whole register.
+        writeln!(f, "\nESR_EL1: {:#010x}", self.0.get())?;
+
+        // Raw print of exception class.
+        writeln!(f, "\r      Exception Class         (EC) : {:#x}", self.0.read(ESR_EL1::EC))?;
+
+        // Exception class.
+        let ec_translation = match self.exception_class() {
+            Some(ESR_EL1::EC::Value::DataAbortCurrentEL) => "Data Abort, current EL",
+            _ => "N/A",
+        };
+        writeln!(f, "\r - {}", ec_translation)?;
+
+        // Raw print of instruction specific syndrome.
+        write!(f, "\r      Instr Specific Syndrome (ISS): {:#x}", self.0.read(ESR_EL1::ISS))
+    }
+}
+
+impl fmt::Display for ExceptionContext {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        writeln!(f, "\r{}", self.esr_el1)?;
+
+        if self.fault_address_valid() {
+            writeln!(f, "\rFAR_EL1: {:#018x}", FAR_EL1.get() as usize)?;
+        }
+
+        writeln!(f, "\r{}", self.spsr_el1)?;
+        writeln!(f, "\rELR_EL1: {:#018x}", self.elr_el1)?;
+        writeln!(f)?;
+        writeln!(f, "\rGeneral purpose register:")?;
+
+        let alternating = |x| -> _ {
+            if x % 2 == 0 { "   " } else { "\r" }
+        };
+
+        // Print two registers per line.
+        for (i, reg) in self.gpr.iter().enumerate() {
+            writeln!(f, "\r      x{: <2}: {: >#018x}{}", i, reg, alternating(i))?;
+        }
+        write!(f, "\r      lr : {:#018x}", self.lr)
+    }
+}
+
+#[no_mangle]
+extern "C" fn current_el0_synchronous(_e: &mut ExceptionContext) {
+    panic!("Should not be here. Use of SP_EL0 in EL1 is not supported.")
+}
+
+#[no_mangle]
+extern "C" fn current_el0_irq(_e: &mut ExceptionContext) {
+    panic!("Should not be here. Use of SP_EL0 in EL1 is not supported.")
+}
+
+#[no_mangle]
+extern "C" fn current_el0_serror(_e: &mut ExceptionContext) {
+    panic!("Should not be here. Use of SP_EL0 in EL1 is not supported.")
+}
```

Comment:
use "BUG: ..." instead of "Should not be here". It's clearer and more consistent with the rest of Theseus.